akka-stream/src/main/scala/akka/stream/impl/StreamOfStreamProcessors.scala

```diff
@@ -82,9 +82,11 @@ private[akka] object MultiStreamOutputProcessor {
     private def closePublisher(withState: CompletedState): Unit = {
       subscriptionTimeout.cancel()
       state.getAndSet(withState) match {
-        case Attached(sub)     ⇒ closeSubscriber(sub, withState)
         case _: CompletedState ⇒ throw new IllegalStateException("Attempted to double shutdown publisher")
-        case Open              ⇒ // No action needed
+        case Attached(sub) ⇒
+          if (subscriber eq null) tryOnSubscribe(sub, CancelledSubscription)
```

Comment:
This will execute when the actor has an enqueued subscriber (Attached) which it didn't yet handle (subscriber eq null), so it needs to onSubscribe first. 

What seems to be missing though if after closePublisher the actor does not stop, it *will* handle the subscribe message again, so I should put there a guard (checking for completion).